docs/features/logforwarding/outputs/splunk-forwarding.adoc

```diff
@@ -43,5 +49,113 @@ spec:
       outputRefs:
         - splunk-receiver
 ----
+1. `token` - Points to the secret containing the Splunk HEC token used for authenticating requests.
+2. `url` - The base URL of the Splunk instance.
+3. `index` - Optional. The name of the index to send events to. If not specified, the default index defined within Splunk is used. This supports template syntax to allow dynamic per-event values.
+4. `source` - Optional. The source of events sent to this sink. This supports template syntax to allow dynamic per-event values.
+5. `indexedFields` - Optional. Fields to be added to Splunk index.
+6. `compression` - Optional. Compression configuration, available to are: `none`, `gzip`. Default is `none`.
+
+=== `source`
+In Splunk, the `source` field typically identifies the origin of a log event.
+To ensure consistency and meaningful categorization, the `source` value can be dynamically derived from the `log_type` and `log_source` fields,
+following Cluster Log Forwarder's conventions.
+
+Below the table with default value depends on log_type and log_source will be used if not set in configuration.
+
+|===
+| |pass:[Infrastructure Journal <br/><em>log_type:infrastructure<br/>log_source:node</em>]|pass:[Infrastructure/Application Container<br/><em>log_type:infrastructure\|application<br/>log_source:container</em>]|pass:[Audit<br/><em>log_type:audit<br/>log_source:auditd\|ovn\|openshiftAPI\|kubeAPI<em/>]|Note
+|`source`|SYSLOG_IDENTIFIER*|ns_name_podName_containerName|.log_source|
+|===
+
+=== `indexed_fields`
+
+In Splunk, `indexed_fields` are fields that are extracted at *index time*, rather than at *search time*. This means their values are stored directly in the index alongside the raw event data, allowing for *faster search performance* on those fields.
+
+However, because `indexed_fields` increase storage usage, they should be used *sparingly* and only for *high-value fields* that provide significant search benefits.
+
+Since logs processed by Vector are typically *well-structured JSON*, Splunk can dynamically extract fields at search time. As a result, the use of `indexed_fields` is *not necessary by default*. That said, they may be valuable in *specific use cases*, such as *large datasets with frequent queries on specific fields*.
+In such cases, users can configure the required fields manually.
+
+==== Limitations
+
+* Only *top-level fields* are supported.
```

Comment:
If we are adding templating which means we will copy a value to the root, there is no need to specify that here

Reply:
I think we need to add some information why we add such remapping

Reply:
From a user's perspective, our API is a "path" which does not necessarily have to be at the root.  We intend to flatten the path to be the key name with underscores.  ClusterLogForwarder this statement is inaccurate even though the implementation requires it at the root